### PR TITLE
[tune] Add `Trainable.postprocess_checkpoint`

### PR DESCRIPTION
**Pull request by Yard1**
## Why are these changes needed?

Adds `postprocess_checkpoint` method to `Trainable` to facilitate the checkpointing of preprocessors in AIR.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/trainable.py [hunk 1]
```diff
@@ -442,11 +442,18 @@ def save(self, checkpoint_dir=None) -> str:
...
         # Maybe sync to cloud
         self._maybe_save_to_cloud(checkpoint_dir)
 
         return checkpoint_path
 
+    def postprocess_checkpoint(self, checkpoint_path: str):
+        """Run any additional postprocessing before the checkpoint
+        is saved to cloud."""
+        pass
+
```
Comment (krfricke) [inline python/ray/tune/trainable.py:456]: Let's make this an internal API for now (then we can change the implementation later)

**Review by krfricke — CHANGES_REQUESTED** [review]
Minor API change

**Review by krfricke — APPROVED** [review]
LGTM

File: python/ray/tune/trainable.py [hunk 2]
```diff
@@ -442,11 +442,18 @@ def save(self, checkpoint_dir=None) -> str:
...
 
+        self._postprocess_checkpoint(checkpoint_path)
+
         # Maybe sync to cloud
         self._maybe_save_to_cloud(checkpoint_dir)
 
         return checkpoint_path
 
+    def _postprocess_checkpoint(self, checkpoint_path: str):
+        """Run any additional postprocessing before the checkpoint
```
Comment (amogkam) [inline python/ray/tune/trainable.py:453]: The first line in the docstring should only be 1 line?

File: python/ray/tune/trainable.py [hunk 3]
```diff
@@ -442,11 +442,18 @@ def save(self, checkpoint_dir=None) -> str:
...
+        is saved to cloud."""
```
Comment (amogkam) [inline python/ray/tune/trainable.py:454]: Should we also document the arg?

Comment (Yard1) [reply python/ray/tune/trainable.py:454]: It's internal, so I don't think it's necessary

File: python/ray/tune/trainable.py (hunk 2 above)
Comment (Yard1) [reply python/ray/tune/trainable.py:453]: Right, it could fit

**Review by amogkam — APPROVED** [review]
LGTM!

**Pull request merged**